### ISSUE TITLE
templates API

### ISSUE THREAD

**Issue by eroux** [BuddhistDigitalResourceCenter/lds-pdi#27]
A few things to fix on the /resource/templates/ API:

- I think it would be better named `template` (no `s`)
- instead of passing the template name in the `?searchType` argument, it could be in the url: `/resource/templates?searchType=IIIFPres_itemInfo` could be `/resource/template/IIIFPres_itemInfo`
- `/resource/template` (no parameter) should return the list of templates
- everything should work with json parameters (in POST or GET)

**Comment by eroux** [comment]
In fact, I think maybe using `/query` instead of `/resource/template` would make more sense, I don't think the `/resource` prefix is that relevant here... there is a little bit of API design to do here, creating a swagger file first will probably help

**Comment by eroux** [comment]
Also, http://buda1.bdrc.io/queries returns links with the `.arq` suffix, which don't work... but using some best practices like [this one](http://51elliot.blogspot.fr/2014/06/rest-api-best-practices-4-collections.html) would make more sense anyways, the output would look something like:

```json
[
  {
    "id": "PersonNames_byName",
    "href": "/queries/PersonNames_byName",
    "description": "Gets some basic info on persons with names matching the argument"
  },
...
]
```

(note that this would require a "description" field on the queries, which would be very useful)

**Comment by xristy** [comment]
+1 These are great suggestions.

**Comment by MarcAgate** [comment]
All points implemented as of commit 6961186 (Except for swagger file which requires some thinking and time)

**Comment by eroux** [comment]
looks good, thanks! the swagger file has its own issue, closing this one
